examples/scan/scan.py

```diff
@@ -0,0 +1,128 @@
+import asyncio
+from datetime import datetime
+import logging
+import pathlib
+import re
+import sys
+from typing import List
+
+from yapapi import (
+    Golem,
+    Task,
+    WorkContext,
+)
+from yapapi.log import pluralize
+from yapapi.payload import vm
+from yapapi.strategy import MarketStrategy, SCORE_REJECTED, SCORE_TRUSTED
+
+examples_dir = pathlib.Path(__file__).resolve().parent.parent
+sys.path.append(str(examples_dir))
+
+from utils import (
+    build_parser,
+    TEXT_COLOR_CYAN,
+    TEXT_COLOR_DEFAULT,
+    TEXT_COLOR_YELLOW,
+    TEXT_COLOR_MAGENTA,
+    run_golem_example,
+    print_env_info,
+)
+
+logger = logging.getLogger(__name__)
+
+# alpine:latest
+IMAGE_HASH = "d646d7b93083d817846c2ae5c62c72ca0507782385a2e29291a3d376"
+
+nodes_list = {}
+
+
+class ScanStrategy(MarketStrategy):
+    async def score_offer(self, offer, _agreements_pool=None):
+        node_address = offer.issuer
+
+        # reject nodes that we have already scanned
+        if node_address in nodes_list:
+            return SCORE_REJECTED
+
+        return SCORE_TRUSTED
+
+
+async def main(
+    scan_size: int, max_workers: int, subnet_tag, payment_driver=None, payment_network=None
+):
+    payload = await vm.repo(image_hash=IMAGE_HASH)
+
+    async def worker(ctx: WorkContext, tasks):
+        assert ctx.provider_id not in nodes_list
+
+        async for task in tasks:
+            print(
+                f"{TEXT_COLOR_CYAN}"
+                f"Getting info for {ctx.provider_id} (aka {ctx.provider_name})"
+                f"{TEXT_COLOR_DEFAULT}",
+            )
+            script = ctx.new_script()
+
+            future_result = script.run("/bin/cat", "/proc/cpuinfo")
+            yield script
+
+            result = (await future_result).stdout or ""
+            nodes_list[ctx.provider_id] = True
```

Comment:
You can change `nodes_list` type from `Dict[str, bool]` to `Set[str]` and just write `nodes_list.add(ctx.provider_id)` here.

Reply:
indeed 👍 

at the beginning, I think I wanted to put some more logic there but indeed, in the final form, that could be a simple set

Reply:
oh, come on, that's nitpicking ;p ...

Reply:
anyway, fixed